Sources/Sentry/Public/SentrySpan.h

```diff
@@ -1,9 +1,93 @@
-#import <Foundation/Foundation.h>
+#import "SentrySpanContext.h"
+#import <Sentry/Sentry.h>
 
 NS_ASSUME_NONNULL_BEGIN
 
+@class SentryTransaction, SentrySpanId, SentryId, SentryHub;
+
 NS_SWIFT_NAME(Span)
-@protocol SentrySpan <NSObject>
+@interface SentrySpan : SentrySpanContext
+SENTRY_NO_INIT
+
+/**
+ * The timestamp of which the span ended.
+ */
+@property (nullable, nonatomic, strong) NSDate *timestamp;
+
+/**
+ * The start time of the span.
+ */
+@property (nullable, nonatomic, strong) NSDate *startTimestamp;
+
+/**
+ * An arbitrary mapping of additional metadata of the span.
+ */
+@property (nullable, readonly) NSDictionary<NSString *, id> *extras;
+
+/**
+ * Whether the span is finished.
+ */
+@property (readonly) bool isFinished;
+
+/**
+ * Init a SentrySpan with given transaction, traceId, parentSpanId and hub.
+ *
+ * @param transaction The transaction associated with this span.
+ * @param traceId Determines which trace the Span belongs to.
+ * @param parentId Id of a parent span.
+ *
+ * @return SentrySpan
+ */
+- (instancetype)initWithTransaction:(SentryTransaction *)transaction
+                            traceId:(SentryId *)traceId
+                        andParentId:(SentrySpanId *)parentId;
+
+/*
+ Removing SpanContext initializers
+ */
+- (instancetype)initWithSampled:(BOOL)sampled NS_UNAVAILABLE;
+- (instancetype)initWithTraceId:(SentryId *)traceId
```

Comment:
`l`: Docs missing for this one.

Reply:
This initializer was hidden. Does it need more documentation.